store/iavl/proof.go

```diff
@@ -0,0 +1,109 @@
+package iavl
+
+import (
+	"errors"
+	"fmt"
+
+	ics23 "github.com/confio/ics23/go"
+	"github.com/tendermint/tendermint/crypto/merkle"
+)
+
+const ProofOpIAVL = "iavlstore"
+
+type IAVLOp struct {
```

Comment:
Since the key is already embedded in ics23.CommitmentProof, I think we don't need to store it separately. We can pull it out on `GetKey()`.

Then we can either do:

```go
type IAVLOp struct {
 	Proof *ics23.CommitmentProof
}
```

Or maybe just a new-type wrapper:

```go
type IAVLOp *ics23.CommitmentProof
```

Reply:
Also, if it makes it easier, use a different proof type for existence and absence. So you don't have to type switch below. Makes more sense to this code, but not sure in how it is used externally.